### PR TITLE
Add nullable support

### PR DESCRIPTION
**Pull request by n8chur**
Adds support for `x-nullable` on `Metadata`.

### CODE REVIEW

File: Tests/SwaggerParserTests/SwaggerParserTests.swift [hunk 1]
```diff
@@ -30,6 +30,46 @@ class SwaggerParserTests: XCTestCase {
         try validate(that: swagger.definitions, containsTestAllOfChild: "TestAllOfFoo", withPropertyNames: ["foo"])
         try validate(that: swagger.definitions, containsTestAllOfChild: "TestAllOfBar", withPropertyNames: ["bar"])
     }
+    
+    func testNullable() throws {
+        let url = testFixtureFolder.appendingPathComponent("test_nullable.json")
+        let jsonString = try NSString(contentsOfFile: url.path, encoding: String.Encoding.utf8.rawValue) as String
```
Comment (AttilaTheFun) [inline Tests/SwaggerParserTests/SwaggerParserTests.swift]: Can you just use the String(contentsOfFile: ...) initializer?

**Review by AttilaTheFun — CHANGES_REQUESTED** [review]
Looks good! Let's just try to use Swift strings instead of NSString

**Review by AttilaTheFun — APPROVED** [review]
Thanks! I think you should be able to merge now. Please use the `Squash and Merge` option from the dropdown on the PR so we have a clean commit history. Moving forwards I'll make any changes via a pull request even if I just merge it immediately. I'll tag the squashed + merged commit with a new subversion. My usual rule of thumb is that:
Pre v1.0: any point change can be breaking, dot changes are not.

Post v1.0: Version bumps for massive, breaking changes / overhauls (e.g. swift 3 -> 4). Point changes for backwards compatible changes that change the interface. Dot version bumps for bug fixes that don't change the interface.

Does this sound good to you?

**Comment by n8chur** [comment]
> Thanks! I think you should be able to merge now. Please use the Squash and Merge option from the dropdown on the PR so we have a clean commit history. Moving forwards I'll make any changes via a pull request even if I just merge it immediately. I'll tag the squashed + merged commit with a new subversion.

Sounds good, I'll be sure to squash and merge.

> My usual rule of thumb is that:
> Pre v1.0: any point change can be breaking, dot changes are not.
>
> Post v1.0: Version bumps for massive, breaking changes / overhauls (e.g. swift 3 -> 4). Point changes for backwards compatible changes that change the interface. Dot version bumps for bug fixes that don't change the interface.
> 
> Does this sound good to you?

Yeah, that sounds good to me! I'm a big fan of using the [semver](http://semver.org) spec as a reference when deciding how to version a release.

**Comment by AttilaTheFun** [comment]
Yeah that sounds good. We should adopt Semver fully after we feel it has reached a "1.0" status. I want to add a README and some contribution guidelines as well.

**Pull request merged**